net/devif/devif_iobsend.c

```diff
@@ -56,67 +57,53 @@ void devif_iob_send(FAR struct net_driver_s *dev, FAR struct iob_s *iob,
                     unsigned int len, unsigned int offset,
                     unsigned int target_offset)
 {
-#ifndef CONFIG_NET_IPFRAG
-  unsigned int limit = NETDEV_PKTSIZE(dev) -
-                       NET_LL_HDRLEN(dev) - target_offset;
+  int ret;
 
-  if (dev == NULL || len == 0 || len > limit)
-#else
   if (dev == NULL || len == 0)
-#endif
     {
-      if (dev->d_iob == NULL)
-        {
-          iob_free_chain(iob);
-        }
+      ret = -EINVAL;
```

Comment:
Case dev==NULL shouldn't it return -ENODEV instead of -EINVAL?

Reply:
Done